src/components/Swap/Summary/index.tsx

```diff
@@ -91,13 +89,44 @@ interface SummaryDialogProps {
 }
 
 export function SummaryDialog(props: SummaryDialogProps) {
+  const [ackPriceImpact, setAckPriceImpact] = useState(false)
+  const [showSpeedbump, setShowSpeedbump] = useState(props.impact?.warning === 'error')
+
+  const onAcknowledgePriceImpact = useCallback(() => {
+    setAckPriceImpact(true)
+    setShowSpeedbump(false)
+  }, [])
+
+  const onAcknowledgeNewTrade = useCallback(() => {
+    if (!showSpeedbump && !ackPriceImpact && props.impact?.warning === 'error') {
+      setShowSpeedbump(true)
+    }
+  }, [ackPriceImpact, props.impact?.warning, showSpeedbump])
+
+  useEffect(() => {
+    if (showSpeedbump && props.impact?.warning !== 'error') {
+      setShowSpeedbump(false)
+    }
+  }, [ackPriceImpact, props.impact, showSpeedbump])
+
   return (
     <>
-      <Header title={<Trans>Review swap</Trans>} />
-      <Body flex align="stretch">
-        <Details {...props} />
-      </Body>
-      <ConfirmButton {...props} highPriceImpact={props.impact?.warning === 'error'} />
+      {!showSpeedbump ? (
+        <>
+          <Header title={<Trans>Review swap</Trans>} />
+          <Body flex align="stretch">
+            <Details {...props} />
+          </Body>
+          <ConfirmButton {...props} onAcknowledgeNewTrade={onAcknowledgeNewTrade} />
+        </>
+      ) : (
+        props.impact && (
+          <SpeedbumpDialog onAcknowledge={onAcknowledgePriceImpact} impact={props.impact}>
```

Comment:
this boolean logic seems like it has the possibility to render `null` - can we combine the two checks to we always fallback to the non-warning view?

like: `props.impact && showSpeedbump ? {speedbump} : {normal view}`